spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/BaseReader.java

```diff
@@ -111,8 +136,11 @@ public boolean next() throws IOException {
         }
       }
     } catch (IOException | RuntimeException e) {
-      if (currentTask != null && !currentTask.isDataTask()) {
```

Comment:
Are we sure we want to change `!currentTask.isDataTask()`?

On a side note, this can be rewritten with the new abstract method:

```
if (currentTask != null && !currentTask.isDataTask()) {
  String filePaths = referencedFiles(currentTask)
      .map(file -> file.path().toString())
      .collect(Collectors.joining(", "));
  LOG.error("Error reading file(s): {}", filePaths, e);
}
```